### PR TITLE
intersect: Add precise line-bezier intersection

### PR DESCRIPTION
**Pull request by johannes-wolf**
Use this for the border anchor calculation. Path/path intersection still defaults to the sampling based linear intersection.

### CODE REVIEW

File: src/intersection.typ [hunk 1]
```diff
@@ -36,6 +36,49 @@
...
+  }
+  return pts
+}
+
+/// Check for line-path intersection in 2D
+///
+/// - la (vector): Line start
+/// - lb (vector): Line end
+/// - path (path): Path
+/// - samples (int): Number of samples per curve
```
Comment (fenjalien) [inline src/intersection.typ:62]: samples doesn't exist

File: src/intersection.typ [hunk 2]
```diff
@@ -36,6 +36,49 @@
   }
 }
 
+// Check for line-cubic bezier intersection
+#let line-cubic(la, lb, s, e, c1, c2) = {
+  import "/src/bezier.typ"
+  return bezier.line-cubic-intersections(la, lb, s, e, c1, c2)
+}
```
Comment (fenjalien) [inline src/intersection.typ:43]: Why not rename this function on import?
```
#import "/src/bezier.typ": line-cubic-intersections as line-cubic

**Review by fenjalien — APPROVED** [review]
this is cool, nice work. Could `path-path` be improved by using `line-path` somehow?

**Comment by johannes-wolf** [comment]
Changed. Also found a little bug in the intersection code :)

**Pull request merged**